models/query_class.go

```diff
@@ -0,0 +1,441 @@
+	:m_bytes_sent_min,
+	:m_bytes_sent_max,
+	:m_bytes_sent_p99,
+	:m_bytes_sent_hg,
+	:m_tmp_tables_cnt,
+	:m_tmp_tables_sum,
+	:m_tmp_tables_min,
+	:m_tmp_tables_max,
+	:m_tmp_tables_p99,
+	:m_tmp_tables_hg,
+	:m_tmp_disk_tables_cnt,
+	:m_tmp_disk_tables_sum,
+	:m_tmp_disk_tables_min,
+	:m_tmp_disk_tables_max,
+	:m_tmp_disk_tables_p99,
+	:m_tmp_disk_tables_hg,
+	:m_tmp_table_sizes_cnt,
+	:m_tmp_table_sizes_sum,
+	:m_tmp_table_sizes_min,
+	:m_tmp_table_sizes_max,
+	:m_tmp_table_sizes_p99,
+	:m_tmp_table_sizes_hg,
+	:m_qc_hit_sum,
+	:m_full_scan_sum,
+	:m_full_join_sum,
+	:m_tmp_table_sum,
+	:m_tmp_table_on_disk_sum,
+	:m_filesort_sum,
+	:m_filesort_on_disk_sum,
+	:m_select_full_range_join_sum,
+	:m_select_range_sum,
+	:m_select_range_check_sum,
+	:m_sort_range_sum,
+	:m_sort_rows_sum,
+	:m_sort_scan_sum,
+	:m_no_index_used_sum,
+	:m_no_good_index_used_sum,
+	:grpstr,
+	:grpint,
+	:labint_key
+	:labint_value
+  )
+`
+
+type QueryClass struct {
+	db *sqlx.DB
+}
+
+func NewQueryClass(db *sqlx.DB) QueryClass {
+	return QueryClass{db: db}
+}
+
+func (qc *QueryClass) Save(agentMsg *collectorpb.AgentMessage) error {
+	type QueryClass struct {
+		PeriodStart   time.Time `json:"period_start_ts"`
+		ExampleType   string    `json:"example_type_s"`
+		ExampleFormat string    `json:"example_format_s"`
+		LabelsKey     []string  `json:"labels_key"`
+		LabelsValues  []string  `json:"labels_value"`
+		WarningsCode  []string  `json:"warnings_code"`
+		WarningsCount []uint64  `json:"warnings_count"`
+		ErrorsCode    []string  `json:"errors_code"`
+		ErrorsCount   []uint64  `json:"errors_count"`
+		LabintKey     []uint32  `json:"labint_key"`
+		LabintValue   []uint32  `json:"labint_value"`
+		*collectorpb.QueryClass
+	}
+
+	qc.db.Mapper = reflectx.NewMapperTagFunc("json", strings.ToUpper, func(value string) string {
+		if strings.Contains(value, ",") {
+			return strings.Split(value, ",")[0]
+		}
+		return value
+	})
+
+	tx, err := qc.db.Beginx()
+	if err != nil {
+		return fmt.Errorf("begin transaction: %s", err.Error())
+	}
+
+	stmt, err := tx.PrepareNamed(insertSQL)
+	if err != nil {
+		return fmt.Errorf("prepare named: %s", err.Error())
+	}
+
+	var errs error
+	for _, qc := range agentMsg.QueryClass {
+		lk, lv := MapToArrsStrStr(qc.Labels)
+		wk, wv := MapToArrsStrInt(qc.Warnings)
+		ek, ev := MapToArrsStrInt(qc.Errors)
+		labintk, labintv := MapToArrsIntInt(qc.Labint)
+		q := QueryClass{
+			time.Unix(int64(qc.GetPeriodStart()), 0),
+			qc.GetExampleType().String(),
+			qc.GetExampleFormat().String(),
+			lk,
+			lv,
+			wk,
+			wv,
+			ek,
+			ev,
+			labintk,
+			labintv,
+			qc,
+		}
+		_, err := stmt.Exec(q)
+
+		if err != nil {
+			errs = fmt.Errorf("%v; execute: %v;", errs, err)
+		}
+	}
+	if errs != nil {
+		return errs 
+	}
+
+	err = stmt.Close()
+	if err != nil {
+		return fmt.Errorf("close statement: %s", err.Error())
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return fmt.Errorf("transaction commit %s", err.Error())
+	}
+
+	return nil
+}
```

Comment:
Let's add a few basic tests for that method: save and read, save partial data and read.